### ISSUE TITLE
Broken access from IPv6

### ISSUE THREAD

**Issue by inetol** [vikejs/vike#1808]
### Description

IPv6 access does not seem to work in specific scenarios...

```ts
589 |   if (!host)
590 |     return ret;
591 |   const [hostname, ...rest] = host.split(":");
592 |   ret.hostname = hostname;
593 |   if (rest.length > 0) {
594 |     assert(rest.length === 1, url);
          ^
error: [vike@0.4.184][Bug] You stumbled upon a Vike bug. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error. A maintainer will fix the bug (usually under 24 hours). Debug info (for Vike maintainers; you can ignore this): http://[::1]:3000/
      at parseHost (/dist/server/index.mjs:594:5)
      at parseUrl (/dist/server/index.mjs:892:16)
      at hasSuffix (/dist/server/index.mjs:42921:42)
      at handlePageContextRequestUrl (/dist/server/index.mjs:42915:8)
      at checkBaseUrl (/dist/server/index.mjs:43638:50)
      at /dist/server/index.mjs:43310:37
GET - / failed
```

I was not able to replicate the issue in a clean environment (scaffolding with Bati), but I can replicate it in a specific repository (Bun required):

1. Clone this repo:
- 1. `git clone https://github.com/jspaste/frontend.git issueIP6`
- 2. `cd issueIP6/`
- 3. `git reset --hard dcaa09e1aa3be05993d73e12b19292b1d122f21b`

2. Install deps: `bun i`
3. Build the frontend: `bun vite build`
4. Run the server: `NODE_ENV=production bun run --cwd=./dist/ ./server/index.mjs`
5. Access to localhost IPv6: *[`http://[::1]:3000`](http://[::1]:3000)*

**Comment by brillout** [comment]
Can you publish a minimal reproduction?

**Comment by inetol** [comment]
https://github.com/inetol/vikejs-vike-1808 ; I was able to reproduce it as well using Hono

**Comment by brillout** [comment]
[Fix](https://github.com/vikejs/vike/commit/7048cc125bd1a3c783ca1e676aac3b6483fc0417) pre-released as `0.4.184-commit-7048cc1`. I tried against your reproduction and it now works.

Btw. would your company be up for sponsoring (#1350)? It makes a massive difference to us!
